scripts/quests/otherAreas/Fishermans_Heart.lua

```diff
@@ -0,0 +1,147 @@
+                end,
+            },
+        },
+    },
+
+    -- Quest Accepted
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Katsunaga'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(194, xi.item.GUGRU_TUNA)
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, xi.item.GUGRU_TUNA) then
+                        local fishingStats = player:getFishingStats() or {}
+                        return quest:progressEvent(193, {
+                            [1] = fishingStats['fishLinesCast'] or 0,
+                            [2] = fishingStats['fishReeled'] or 0,
+                            [3] = fishingStats['fishLongestLength'] or 0,
+                            [4] = fishingStats['fishLongestId'] or 0,
+                            [5] = fishingStats['fishHeaviestWeight'] or 0,
+                            [6] = fishingStats['fishHeaviestId'] or 0,
+                        })
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [193] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    quest:complete(player)
+                end,
+            },
+        },
+    },
+
+    -- Quest Complete
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_COMPLETED
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Katsunaga'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(197, player:getFishingCatchCount() or 0) -- Number of fish types caught
+                end,
+            },
+
+            onEventUpdate =
+            {
+                [197] = function(player, csid, option, npc)
+                    if option == 33554432 then
+                        -- 'Catching Fish'
+                        local fishingStats = player:getFishingStats() or {}
+                        player:updateEvent({
+                            [0] = player:getFishingCatchCount(),
+                            [1] = fishingStats['fishLinesCast'] or 0,
+                            [2] = fishingStats['fishReeled'] or 0,
+                            [3] = fishingStats['fishLongestLength'] or 0,
+                            [4] = fishingStats['fishLongestId'] or 0,
+                            [5] = fishingStats['fishHeaviestWeight'] or 0,
+                            [6] = fishingStats['fishHeaviestId'] or 0,
+                        })
+                    elseif option == 0 then
+                        -- 'Types of Fish Caught'
+                        local fishingCatches = player:getFishingCatches() or {}
```

Comment:
Need to check this datatype to see if it can be simplified, similar to above

Reply:
The 24-byte bitset (for 196 bits) uses 163 of those bits to build a set of 6 32-bit integers as flags for which fish have been caught.  I'm not sure if that's actually _the_ easiest method, but it is how the client receives the data to show "which fish should I put a star next to?"